modules/aws/iam.go

```diff
@@ -135,26 +140,43 @@ func EnableMfaDeviceE(t *testing.T, iamClient *iam.IAM, mfaDevice *iam.VirtualMF
 
 // AssertIamPolicyExists checks if the given IAM policy exists in the given region and fail the test if it does not.
 func AssertIamPolicyExists(t *testing.T, region string, policyARN string) {
-	err := AssertIamPolicyExistsE(t, region, policyARN)
+	_, err := GetIamPolicyDocumentE(t, region, policyARN)
 	if err != nil {
 		t.Fatal(err)
 	}
 }
 
-//AssertIamPolicyExistsE checks if the given IAM policy exists in the given region and return an error if it does not.
-func AssertIamPolicyExistsE(t *testing.T, region string, policyARN string) error {
-	iamClient, err := NewIamClientE(t, region)
+// AssertIAMPolicyIsCorrect fetches the contents of the bucket policy for the given bucket and matches it against the provided policy document.
```

Comment:
This seems a bit confusing. IAM policy != S3 Bucket policy.

Reply:
todays lesson, don't copy-paste so much